File: frontend/src/api/analysis.ts
import client from "./client";

export interface AnalysisResponse {
  task_id: string;
  status: string;
  estimated_seconds: number;
}

export interface AnalysisStatus {
  task_id: string;
  stock_code: string;
  stock_name: string | null;
  status: string;
  progress: number;
  report: string | null;
  html_report: string | null;
  error: string | null;
  created_at: string;
  updated_at: string;
}

export interface HistoryItem {
  task_id: string;
  stock_code: string;
  stock_name: string | null;
  status: string;
  created_at: string;
}

export interface HistoryResponse {
  total: number;
  page: number;
  page_size: number;
  items: HistoryItem[];
}

export const analysisApi = {
  start(stockCode: string, email: string, skillName = "cninfo-financial-analysis"): Promise<AnalysisResponse> {
    return client.post("/analysis/start", { stock_code: stockCode, email, skill_name: skillName }).then((r) => r.data);
  },

  getStatus(taskId: string): Promise<AnalysisStatus> {
    return client.get(`/analysis/${taskId}/status`).then((r) => r.data);
  },

  getHistory(page = 1, pageSize = 20): Promise<HistoryResponse> {
    return client.get("/analysis/history", { params: { page, page_size: pageSize } }).then((r) => r.data);
  },
};

File: backend/app/schemas/analysis.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional


class AnalysisRequest(BaseModel):
    stock_code: str = Field(..., min_length=1, max_length=20, pattern=r"^[0-9]+$", description="股票代码，仅允许纯数字，如 600519")
    skill_name: str = Field(default="cninfo-financial-analysis", description="Hermes skill 名称")
    email: str = Field(default="", description="接收报告的邮箱（从JWT自动获取，可留空）")


class AnalysisResponse(BaseModel):
    task_id: str
    status: str
    estimated_seconds: int = 600


class AnalysisStatusResponse(BaseModel):
    task_id: str
    stock_code: str
    stock_name: Optional[str] = None
    status: str
    progress: float
    report: Optional[str] = None
    html_report: Optional[str] = None
    error: Optional[str] = None
    created_at: datetime
    updated_at: datetime


class AnalysisHistoryItem(BaseModel):
    task_id: str
    stock_code: str
    stock_name: Optional[str] = None
    status: str
    created_at: datetime

    class Config:
        from_attributes = True
